### PR TITLE
[WIP][FB] Static Date Picker Component Phase 1

### PR DESCRIPTION
**Pull request by DaniFrancisco**
Feature branch: https://uniplaces.atlassian.net/browse/UN-15026. Do not commit directly on this branch

Merged into this:
- https://uniplaces.atlassian.net/browse/UN-15028 (#290)
- https://uniplaces.atlassian.net/browse/UN-15029 (#292)
- https://uniplaces.atlassian.net/browse/UN-15030 (#293)
- https://uniplaces.atlassian.net/browse/UN-15033 (#296)
- https://uniplaces.atlassian.net/browse/UN-15294 (#297)
- https://uniplaces.atlassian.net/browse/UN-15295 (#299)

### CODE REVIEW

**Comment by migueldemoura** [comment]
@niceoutput rebase please

File: addon/components/uni-datepicker.js [hunk 1]
```diff
@@ -16,16 +21,33 @@ export default Component.extend({
...
   }),
+  selectedMessage: computed('messages', function() {
+    let messagesArray = this.get('messages');
+
+    if (isEmpty(messagesArray)) {
+      return null;
+    }
+
+    return messagesArray.sort(function(a, b) {
+      if (a.type === b.type) {
```
Comment (AndreJoaquim) [inline addon/components/uni-datepicker.js:43]: Why do you have a guard clauses mixed with ternary operators? Please use only guard clauses:

```js
if (a.type === b.type) {
  return 0;
}

if (a.type === MessageType.INFORMATION) {
  return 1;
}

return -1;
```

File: addon/styles/addon.scss [hunk 2]
```diff
@@ -1,4 +1,5 @@
 @import "node_modules/uniplaces-uniq/src/styles/_main";
+@import "node_modules/ember-power-calendar/app/styles/ember-power-calendar";
```
Comment (AndreJoaquim) [inline addon/styles/addon.scss:2]: `@import "ember-power-calendar"`

File: addon/styles/partials/_uni-datepicker.scss [hunk 3]
```diff
@@ -1,57 +1,54 @@
-$box-size: 42px;
+$cell-size: 40px;
+$cell-border-color: #eaebec;
+$day-color: #3d4a52;
+$disabled-background-color: #f3f3f4;
+$text-main-color: #161e22;
```
Comment (AndreJoaquim) [inline addon/styles/partials/_uni-datepicker.scss:5]: These new colors should be on `uniq`, not here -- on the `color('calendar', ...)`

File: addon/templates/components/uni-datepicker.hbs [hunk 4]
```diff
@@ -2,18 +2,36 @@
...
   selected=selected
   onCenterChange=(action onCenter value="moment")
-  onSelect=(action onSelect value="moment") as |calendar|}}
+  onSelect=(action onSelect value="moment")
+  classNames=(if isFullScreen "fullscreen")
+as |calendar|}}
+
+  {{yield}}
+
+  {{#if isFullScreen}}
```
Comment (AndreJoaquim) [inline addon/templates/components/uni-datepicker.hbs:18]: I don't think the `uni-datepicker` should be the one "fullscreening": it should be the uni-calendar :/

File: addon/components/uni-calendar.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import Component from '@ember/component';
+import layout from '../templates/components/uni-calendar';
+
+export default Component.extend({
+  layout,
+
+  showAbove: false
```
Comment (AndreJoaquim) [inline addon/components/uni-calendar.js]: I would suggest `position: 'above' | 'below'` to be more flexible in the future  :)

File: addon/styles/partials/_uni-datepicker.scss [hunk 6]
```diff
@@ -79,124 +80,168 @@ $box-size: 42px;
...
-      &:hover {
-        background-color: color('calendar', 'hover-day-bg') !important;
-      }
+      background-color: transparent !important;
+      border-right: 1px solid $cell-border-color;
 
       &:disabled {
-        color: color('buttons', 'disabled-text') !important;
+        background-color: $disabled-background-color !important;
+        color: #dcdfe1 !important;
```
Comment (migueldemoura) [inline addon/styles/partials/_uni-datepicker.scss:88]: Are all these importants needed?

**Pull request closed without merging**